### PR TITLE
Fixes for JAXRS-CXF Codegen

### PR DESCRIPTION
**Pull request by hiveship**
i'm the original author of the CXF Codegen. I've made some updates. I'm now using this codegen in a project, it's OK.

- Add support for CXF 2 (see notes for the Interface Comparator). This Codegen now supports CXF 2 & 3

- Fix bugs for unmarshalling models (tested with Jackson v1 & v2)

- Fix some other critical bugs (this bugs shows that nobody was actually using this Codegen)

### CODE REVIEW

**Comment by wing328** [comment]
@hiveship your first commit https://github.com/hiveship/swagger-codegen/commit/37e62b89bc7f4b1be2f7efeaf70f1f42dd7ea652 is not linked to your github account and therefore that won't count as your contribution in https://github.com/swagger-api/swagger-codegen/graphs/contributors.

File: README.md [hunk 1]
```diff
@@ -628,6 +628,23 @@ java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate \
...
+
+You must register this class into your JAX-RS configuration file:
+```xml
+         <jaxrs:resourceComparator>
+   			<bean class="com.akio.cim.aip.rest.utils.CXFInterfaceComparator"/>
+		</jaxrs:resourceComparator> 
+```
+
+This is no longer necessary if you are using CXF >=v3.x
+
```
Comment (wing328) [inline README.md]: I found some grammatical mistakes and make an update in another commit.

**Comment by wing328** [comment]
Got an error testing the PR:
```
(venv)swagger-codegen|fix-cxf-codegen ⇒ ./bin/jaxrs-cxf-petstore-server.sh 
reading from modules/swagger-codegen/src/test/resources/2_0/petstore.json
[main] INFO io.swagger.codegen.AbstractGenerator - writing file /private/var/tmp/pr/hiveship/swagger-codegen/samples/server/petstore/jaxrs-cxf/gen/java/io/swagger/model/User.java
[main] INFO io.swagger.codegen.AbstractGenerator - writing file /private/var/tmp/pr/hiveship/swagger-codegen/samples/server/petstore/jaxrs-cxf/gen/java/io/swagger/model/Category.java
[main] ERROR io.swagger.codegen.AbstractGenerator - JavaJaxRS/cxf/enumClass.mustache (No such file or directory)
Exception in thread "main" java.lang.RuntimeException: Could not generate model 'Pet'
	at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:268)
	at io.swagger.codegen.cmd.Generate.run(Generate.java:188)
	at io.swagger.codegen.SwaggerCodegen.main(SwaggerCodegen.java:36)
Caused by: java.lang.RuntimeException: can't load template JavaJaxRS/cxf/enumClass.mustache
	at io.swagger.codegen.AbstractGenerator.getTemplateReader(AbstractGenerator.java:63)
	at io.swagger.codegen.DefaultGenerator$1.getTemplate(DefaultGenerator.java:237)
	at com.samskivert.mustache.Mustache$IncludedTemplateSegment.execute(Mustache.java:663)
	at com.samskivert.mustache.Mustache$BlockSegment.executeSegs(Mustache.java:718)
	at com.samskivert.mustache.Mustache$SectionSegment.execute(Mustache.java:742)
	at com.samskivert.mustache.Mustache$BlockSegment.executeSegs(Mustache.java:718)
	at com.samskivert.mustache.Mustache$SectionSegment.execute(Mustache.java:738)
	at com.samskivert.mustache.Template.executeSegs(Template.java:92)
	at com.samskivert.mustache.Mustache$IncludedTemplateSegment.execute(Mustache.java:674)
	at com.samskivert.mustache.Mustache$BlockSegment.executeSegs(Mustache.java:718)
	at com.samskivert.mustache.Mustache$InvertedSegment.execute(Mustache.java:770)
	at com.samskivert.mustache.Mustache$BlockSegment.executeSegs(Mustache.java:718)
	at com.samskivert.mustache.Mustache$SectionSegment.execute(Mustache.java:753)
	at com.samskivert.mustache.Mustache$BlockSegment.executeSegs(Mustache.java:718)
	at com.samskivert.mustache.Mustache$SectionSegment.execute(Mustache.java:738)
	at com.samskivert.mustache.Template.executeSegs(Template.java:92)
	at com.samskivert.mustache.Template.execute(Template.java:69)
	at com.samskivert.mustache.Template.execute(Template.java:60)
	at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:242)
	... 2 more
```

Please update the Petstore samples for `jaxrs-cxf` as well.

File: README.md (hunk 1 above)
Comment (hiveship) [inline README.md]: OK thank you. I also updated the file to remove my company name in the 'CXFInterfaceComparator' package. Pease, include it.

**Comment by hiveship** [comment]
I've take a look this mornig and I don't see why this bug occurs (CXFCodegen, JAXRSCodegen & Java Codegen don't import 'enumClass.mustache'). I'm ucrrently using this version on some of my projects (that use enums class and I don't have any problem.
I don't have time to investigate more today and I'll correct it as soon as possible.

**Comment by hiveship** [comment]
>  wing328 commented 2 days ago

@hiveship your first commit hiveship@37e62b8 is not linked to your github account and therefore that won't count as your contribution in https://github.com/swagger-api/swagger-codegen/graphs/contributors.
> 

Is that a problem for you ? How can I attach this commit to my account ?

**Comment by hiveship** [comment]
>  wing328 commented 2 days ago
@hiveship your first commit hiveship@37e62b8 is not linked to your github account and therefore that won't count as your contribution in https://github.com/swagger-api/swagger-codegen/graphs/contributors.
> 

Updated.

**Comment by wing328** [comment]
@hiveship please take your time. Let me know after you've updated it and thenI'll give it another try.

**Comment by hiveship** [comment]
@wing328 Could you please retry ? I just added a commit and the sample generation is OK for me.
